### PR TITLE
Bug/2622 pool custom part2

### PR DESCRIPTION
**Pull request by crbrox**
@fgalan

### CODE REVIEW

File: src/lib/ngsi/ContextElementResponseVector.cpp [hunk 1]
```diff
@@ -162,7 +162,7 @@ void ContextElementResponseVector::push_back(ContextElementResponse* item)
 *
```
Comment (fgalan) [inline src/lib/ngsi/ContextElementResponseVector.cpp]: CNR entry is missing. Suggestion:

```
Fix: threadpool mode not actually used for custom notifications, even if -notificationMode is set so (#2622)
```

File: src/lib/ngsiNotify/Notifier.cpp [hunk 2]
```diff
@@ -54,262 +57,304 @@ Notifier::~Notifier (void)
...
+  pthread_t                         tid;
+  std::vector<SenderThreadParams*>  *paramsV = Notifier::buildSenderParams(ncrP, httpInfo, tenant, xauthToken, fiwareCorrelator, renderFormat, attrsOrder, metadataFilter, blackList);
+
+  if (!paramsV->empty()) // al least one param, an empty vector means an error happened
+  {
+    int ret = pthread_create(&tid, NULL, startSenderThread, paramsV);
+    if (ret != 0)
+    {
+      LM_E(("Runtime Error (error creating thread: %d)", ret));
+      for (unsigned ix = 0; ix < paramsV->size(); ix++) {
```
Comment (fgalan) [inline src/lib/ngsiNotify/Notifier.cpp]: Style: `{` in next line

File: src/lib/ngsiNotify/Notifier.cpp [hunk 3]
```diff
@@ -54,262 +57,304 @@ Notifier::~Notifier (void)
...
-    std::string key   = it->first;
-    std::string value = it->second;
 
-    macroSubstitute(&key,   it->first, ce);
-    macroSubstitute(&value, it->second, ce);
 
-    if (key == "")
+/* ****************************************************************************
+*
+* buildSenderParamsFromTemplate -
```
Comment (fgalan) [inline src/lib/ngsiNotify/Notifier.cpp]: It should be changed to th enambe of the function.

File: src/lib/ngsi/ContextElementResponseVector.cpp (hunk 1 above)
Comment (crbrox) [inline src/lib/ngsi/ContextElementResponseVector.cpp]: fixed 8d1cec3

File: src/lib/ngsiNotify/Notifier.cpp (hunk 2 above)
Comment (crbrox) [inline src/lib/ngsiNotify/Notifier.cpp]: fixed 9b2d0ab

**Comment by fgalan** [comment]
LGTM. Waiting for test harness validation to merge

**Pull request merged**